sdks/universal-router-sdk/src/swapRouter.ts

```diff
@@ -43,6 +64,45 @@ export abstract class SwapRouter {
     })
   }
 
+  /**
+   * Builds the call parameters for a migration from a V3 position to a V4 position.
+   * Some requirements of the parameters:
+   *   - v3RemoveLiquidityOptions.collectOptions.recipient must equal v4PositionManager
+   *   - v3RemoveLiquidityOptions.liquidityPercentage must be 100%
+   *   - input pool and output pool must have the same tokens
+   *   - V3 NFT must be approved, or valid inputV3NFTPermit must be provided with UR as spender
```

Comment:
was considering not taking `V3RemoveOptions` and `V4RemoveOptions` and rather taking minimal deduplicated params and handling these automatically.. decided for this approach to give more configurability to options for each stage